src/compiler/compile.js

```diff
@@ -627,7 +628,8 @@ function makeTerminalNodeLinkFn (el, dirName, value, options, def) {
     filters: parsed.filters,
     raw: value,
     // either an element directive, or if/for
-    def: def || publicDirectives[dirName]
+    // #2366 or custom terminal directive
+    def: def || publicDirectives[dirName] || Vue.options.directives[dirName]
```

Comment:
Thanks, this should actually be `def || resolveAsset(options, 'directives', dirName)` so that:

1. We don't need to import `Vue` here (feels dirty)
2. We reuse the same asset resolution logic (auto warnings if no directive found)